libs/execution/include/hpx/execution/executors/parallel_executor.hpp

```diff
@@ -322,8 +322,7 @@ namespace hpx { namespace parallel { namespace execution {
         template <typename Result, typename F, typename Iter, typename... Ts>
         void spawn_hierarchical(std::vector<hpx::future<Result>>& results,
             lcos::local::latch& l, std::size_t base, std::size_t size,
-            std::size_t num_tasks, F const& func, Iter it,
-            Ts const&... ts) const
+            std::size_t num_tasks, F&& func, Iter it, Ts&&... ts) const
         {
```

Comment:
Changing the arguments to forwarding references would require to forward them inside the function body, no?

Reply:
The arguments are used more than once inside the body.